csharp/test/Ice/info/TestI.cs

```diff
@@ -61,26 +40,24 @@ public class TestIntf : ITestIntf
         {
             TestHelper.Assert(current.Connection != null);
             var ctx = new Dictionary<string, string>();
-            ConnectionInfo info = current.Connection.GetConnectionInfo();
-            ctx["adapterName"] = info.AdapterName!;
-            ctx["incoming"] = info.Incoming ? "true" : "false";
+            ctx["adapterName"] = current.Connection.Adapter?.Name ?? "";
+            ctx["incoming"] = current.Connection.IsIncoming ? "true" : "false";
 
-            IPConnectionInfo? ipinfo = getIPConnectionInfo(info);
-            TestHelper.Assert(ipinfo != null);
-            ctx["localAddress"] = ipinfo.LocalAddress;
-            ctx["localPort"] = ipinfo.LocalPort.ToString();
-            ctx["remoteAddress"] = ipinfo.RemoteAddress;
-            ctx["remotePort"] = ipinfo.RemotePort.ToString();
+            var iPConnection = current.Connection as IPConnection;
+            TestHelper.Assert(iPConnection != null);
+            ctx["localAddress"] = iPConnection.LocalAddress?.Address.ToString() ?? "";
```

Comment:
>  Does TestHelper.Assert(ipConnection != null) "help" the compiler just like a Debug.Assert(ipConnection != null)?

Yes it does it works because the `[DoesNotReturnIf(false)] bool b` parameter in its definition

https://github.com/zeroc-ice/ice/blob/master/csharp/test/TestCommon/TestHelper.cs#L33

Reply:
fixed